homeassistant/components/wyoming/devices.py

```diff
@@ -0,0 +1,72 @@
+"""Class to manage satellite devices."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass, field
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import entity_registry as er
+
+from .const import DOMAIN
+
+
+@dataclass
+class SatelliteDevice:
+    """Class to store device."""
+
+    satellite_id: str
+    device_id: str
+    is_active: bool = False
+    is_enabled: bool = True
+    update_listeners: list[Callable[[SatelliteDevice], None]] = field(
+        default_factory=list
+    )
+
+    @callback
+    def set_is_active(self, active: bool) -> None:
+        """Set active state."""
+        self.is_active = active
+        for listener in self.update_listeners:
+            listener(self)
+
+    @callback
+    def set_is_enabled(self, enabled: bool) -> None:
+        """Set enabled state."""
+        self.is_enabled = enabled
```

Comment:
Should this cancel an ongoing run for wake word ?

Reply:
It does in the satellite by slipping an empty audio chunk into the stream (cancelling the pipeline). I've split out the listeners to make this more clear.